src/TheThingsNetwork.h

```diff
@@ -20,6 +20,14 @@
 
 #define TTN_BUFFER_SIZE 300
 
+#if defined(ARDUINO_ARCH_AVR)
+typedef HardwareSerial SerialType;
+#elif defined(ARDUINO_ARCH_SAM) || defined(ARDUINO_ARCH_SAMD)
+typedef Uart SerialType;
+#else
+typedef Stream SerialType;
+#endif
```

Comment:
The code breaks on `Stream`, because there's no `begin()`, right?

We can't introduce breaking changes in this repository.

We also have users using `AltSoftSerial`, deriving from `Stream()`, introducing `begin()`.

Reply:
No in this case because Serial Type is depending on target and all defined target have `begin()`

For AltSoftSerial, it's another thing, I need to think on how to avoid problem with this one.

Reply:
Where's `begin()` in `Stream` defined?

Not here: https://github.com/arduino/Arduino/blob/master/hardware/arduino/avr/cores/arduino/Stream.h, nor in https://github.com/arduino/Arduino/blob/master/hardware/arduino/avr/cores/arduino/Print.h

Reply:
Sure the original TTN code breaks on begin() because it's a Stream. I'm just saying that this PR does not break because there is a `typedef` of `SerialType` depending on target (AVR or SAMD) using this in the include file
```
#if defined(ARDUINO_ARCH_AVR)
typedef HardwareSerial SerialType;
#elif defined(ARDUINO_ARCH_SAM) || defined(ARDUINO_ARCH_SAMD)
typedef Uart SerialType;
#else
typedef Stream SerialType;
#endif
```

and new TTN object instanceuse SerialType
```cpp
TheThingsNetwork(SerialType &modemStream, Stream &debugStream, ttn_fp_t fp, uint8_t sf = TTN_DEFAULT_SF, uint8_t fsb = TTN_DEFAULT_FSB);
```
I need to see what we can do for AltSoftSerial, but I do not see any other solution than a define.